nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/client/nifi/ControllerClient.java

```diff
@@ -61,6 +66,26 @@ public interface ControllerClient {
     VersionedReportingTaskImportResponseEntity importReportingTasks(VersionedReportingTaskImportRequestEntity importRequestEntity)
             throws NiFiClientException, IOException;
 
+    FlowAnalysisRulesEntity getFlowAnalysisRules() throws NiFiClientException, IOException;
+
+    FlowAnalysisRuleEntity getFlowAnalysisRule(final String id) throws NiFiClientException, IOException;
+
+    PropertyDescriptorEntity getFlowAnalysisRulePropertyDescriptor(final String componentId, final String propertyName, final Boolean sensitive) throws NiFiClientException, IOException;
```

Comment:
This method is called from a test only.

Reply:
Yes, and as I understand it it's normal to have API methods for system test purposes only.